src/contracts/core/StakeRootCompendium.sol

```diff
@@ -0,0 +1,618 @@
+        require(
+            depositInfo.balance
+                >= minDepositBalance(
+                    operatorSetToStrategyAndMultipliers[operatorSet.avs][operatorSet.operatorSetId].length()
+                ),
+            InsufficientDepositBalance()
+        );
+    }
+
+    /// @inheritdoc IStakeRootCompendium
+    function addOrModifyStrategiesAndMultipliers(
+        uint32 operatorSetId,
+        StrategyAndMultiplier[] calldata _strategiesAndMultipliers
+    ) external {
+        OperatorSet memory operatorSet = OperatorSet({avs: msg.sender, operatorSetId: operatorSetId});
+        // update the deposit balance for the operator set whenever number of strategies is changed
+        _updateDepositInfo(operatorSet);
+
+        EnumerableMap.AddressToUintMap storage strategiesAndMultipliers =
+            operatorSetToStrategyAndMultipliers[operatorSet.avs][operatorSet.operatorSetId];
+
+        uint256 numStrategiesBefore = strategiesAndMultipliers.length();
+        // set the strategies and multipliers for the operator set
+        for (uint256 i = 0; i < _strategiesAndMultipliers.length; i++) {
+            strategiesAndMultipliers.set(
+                address(_strategiesAndMultipliers[i].strategy), uint256(_strategiesAndMultipliers[i].multiplier)
+            );
+        }
+        uint256 numStrategiesAfter = strategiesAndMultipliers.length();
+
+        DepositInfo storage depositInfo = depositInfos[operatorSet.avs][operatorSet.operatorSetId];
+
+        // make sure they have enough to pay for MIN_PREPAID_PROOFS
+        require(depositInfo.balance >= minDepositBalance(numStrategiesAfter), InsufficientDepositBalance());
+        // note that they've shown increased demand for proofs
+        depositInfo.lastDemandIncreaseTimestamp = uint32(block.timestamp);
+
+        // only adding new strategies to count
+        _updateTotalStrategies(numStrategiesBefore, numStrategiesAfter);
+    }
+
+    /// @inheritdoc IStakeRootCompendium
+    function removeStrategiesAndMultipliers(uint32 operatorSetId, IStrategy[] calldata strategies) external {
+        OperatorSet memory operatorSet = OperatorSet({avs: msg.sender, operatorSetId: operatorSetId});
+        // update the deposit balance for the operator set whenever number of strategies is changed
+        _updateDepositInfo(operatorSet);
+
+        EnumerableMap.AddressToUintMap storage strategiesAndMultipliers =
+            operatorSetToStrategyAndMultipliers[operatorSet.avs][operatorSetId];
+
+        // note below either all strategies are removed or none are removed and transaction reverts
+        uint256 numStrategiesBefore = strategiesAndMultipliers.length();
+        for (uint256 i = 0; i < strategies.length; i++) {
+            require(strategiesAndMultipliers.remove(address(strategies[i])), NonexistentStrategy());
+        }
+
+        _updateTotalStrategies(numStrategiesBefore, numStrategiesBefore - strategies.length);
```

Comment:
We only care about demand increases in storage since withdrawability is pushed further out in the future by increases in demand do offchain sotware doesn't scale and then get rugged